mrp_production_putaway_strategy/models/mrp_production.py

```diff
@@ -7,22 +7,27 @@
 class MrpProduction(models.Model):
     _inherit = "mrp.production"
 
-    @api.model
-    def create(self, vals):
-        location_dest = self.env["stock.location"].browse(vals.get("location_dest_id"))
-        product = self.env["product.product"].browse(vals.get("product_id"))
-        location_id = location_dest._get_putaway_strategy(product)
-        if location_id:
-            vals["location_dest_id"] = location_id.id
-        mo = super().create(vals)
-        if location_id:
+    @api.model_create_multi
+    def create(self, vals_list):
+        for vals in vals_list:
+            location_dest = self.env["stock.location"].browse(
+                vals.get("location_dest_id")
+            )
+            product = self.env["product.product"].browse(vals.get("product_id"))
+            location_id = location_dest._get_putaway_strategy(product)
+            if location_id:
+                vals["location_dest_id"] = location_id.id
+
+        mos = super().create(vals_list)
+
+        for mo in mos:
             message = (
                 _(
-                    "Applied Putaway strategy to finished products.\n Finished "
-                    "Products Location: %s. "
+                    "Applied Putaway strategy to finished products.\n"
+                    "Finished Products Location: %s."
```

Comment:
This is not correct, you cannot post the message in all MOs that are created, it should be posted when the putaway was actually applied. For example in runbot, without any putaway, I create a MO and the message is added to the chatter:

<img width="1849" height="745" alt="image" src="https://github.com/user-attachments/assets/cbcc2d2d-6100-4273-8ebb-b7263fd83e01" />